pkg/cvo/cvo.go

```diff
@@ -93,6 +93,9 @@ type Operator struct {
 	// minimumUpdateCheckInterval is the minimum duration to check for updates from
 	// the upstream.
 	minimumUpdateCheckInterval time.Duration
+	// architecture identifies the architecture of the cluster nodes. The cluster architecture
+	// is heterogeneous if the cluster has nodes running on different architectures.
```

Comment:
> ... if the release image declares 'release.openshift.io/architecture: multi'.

right?  Because you can be, and ask for, heterogeneous release images without actually having heterogeneous nodes.  And the homogeneous -> heterogeneous transition will (eventually) be:

1. Running a homogeneous release on a homogeneous cluster, asking for homogeneous update recommendations.
2. Running a homogeneous release on a homogeneous cluster, asking for heterogeneous update recommendations.
3. Updating to a heterogeneous release on a homogeneous cluster, asking for heterogeneous update recommendations.
4. Running a heterogeneous release on a homogeneous cluster, asking for heterogeneous update recommendations.
5. Running a heterogeneous release on a heterogenous cluster, asking for heterogeneous update recommendations.

Reply:
Yeah, agree, but that's the "to be" logic. Implementation of this Jira is just a baby step so I made the comment only reflect what this code does now. This value simply gets to set to either "mullti" if the payload contains it or GOARCH. We can expand when we add more logic.

Reply:
Updated comment